packages/ListBox/src/components/Option/helpers/optionState.js

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import uuidv4 from "uuid/v4";
+
+export function createOption({ index, child, groupLabel = null, groupId = null }) {
+  const { isDisabled, isHidden, isSelected, label, onClick, preventDefaultOnSelect, value } = child.props;
+
+  return {
+    content: child,
+    groupId,
+    groupLabel,
+    hasLabel: label,
+    id: `opt_${uuidv4()}`,
+    index,
+    isDisabled,
+    isHidden,
+    isSelected,
+    label: label || child.props.children, // we will try to extract the label from the children if doesn't have label
+    onClick,
+    preventDefaultOnSelect,
+    value: value || undefined,
+  };
+}
+
+export function getDataOptions(children) {
+  if (!children) throw Error("Listbox.Options is a required prop, please check you are passing correctly the data");
+  if (!children.length) return [];
+
+  const options = {};
+  let index = 0;
+
+  React.Children.toArray(children).forEach(child => {
+    if (
+      child.type &&
+      (child.type.componentType === "ListBox.Option" || child.type.componentType === "ListBox.RawItem")
+    ) {
+      options[index] = createOption({ index, child, groupId: child.props.groupId });
+      index += 1;
+    }
+
+    return child;
+  });
+
+  return options;
```

Comment:
Why isn't this an array?

Reply:
because is easier to access options[index] than `options.find()`

Reply:
Mmmm I have to think about the nuances of transforming the option to an array, not sure yet it would work better, but this is something that doesn't affect the API and is an internal we can revisit as we keep improving this component.